source/actions3/actions/AMScanAction.h

```diff
@@ -39,42 +39,95 @@ class AMScanController;
 #define AMSCANACTION_INVALILD_NO_VALID_ACTION_INFO 103112
 #define AMSCANACTION_INVALILD_BEAMLINE_CLAIMS_INVALID 103113
 
+/*!
+  * A class which adapts between an AMScanController and the AMAction3 system.
+  * State instructions passed to this class will be translated and passed on down
+  * to the controller, when relevant. Conversely relevant transition signal from
+  * the controller will trigger the translated state transitions from the Action3
+  * interface in this class.
+  */
 class AMScanAction : public AMAction3
 {
 	Q_OBJECT
 
 public:
-	/// Constructor.  Takes an AMScanConfiguration \param config and builds a scan action around it.  This will create a scan controller which can be controlled from within the workflow.
+	/*!
+	  * Creates an instance of a scan action with the provided scan action info.
+	  * The created scan action will not have a valid scan controller.
+	  */
+
 	Q_INVOKABLE AMScanAction(AMScanActionInfo *info, QObject *parent = 0);
-	/// Copy constructor.
+
+	/*!
+	  * Creates a shallow copy of the provided other scan action, but with no
+	  * valid scan controller set.
+	  * //// Is this intended ^ or am I mistaken about the shallow part. The info
+	  * is stored as a reference so we'll just copy the reference, no? ////
+	  */
```

Comment:
Although I've talked to @iainworkman about this already.  It would be good for the @acquaman/acquaman-working-group knows this as well.  Although we follow the same syntax format as real copy constructors, our copy constructors are __fake__.  Except for a few of our non-QObject classes, the copy constructor is a function that looks like a copy constructor so that our ```createCopy()``` methods can be as simple as possible.